syntax = "proto3";

package user_service;

message User {
  string email = 1;
  string password = 2;
  string name = 3;
  repeated Curso cursos = 4;
  int32 admin = 5;
}

message Curso {
  string id = 1;
  string nombre = 2;
  string descripcion = 3;
}

message RegisterRequest {
  User user = 1;
}

message EliminarMatriculaRequest {
    string email = 1;      
    string course_id = 2;   
}

message EliminarMatriculaResponse{
  string message = 1;
  bool success = 2;
}
message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
  User content = 3; 
}
message EliminarUserRequest {
  string email = 1;
}

message EliminarUserResponse {
  bool success = 1;
  string message = 2;
}

message ObtenerUserRequest {
  string email = 1;
}

message ObtenerUserResponse {
  int32 admin = 1;
}

message ObtenerUsersRequest {}

message ObtenerUsersResponse {
  repeated User users = 1;
}

service UserService {
  rpc Register (RegisterRequest) returns (RegisterResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc EliminarUser (EliminarUserRequest) returns (EliminarUserResponse);
  rpc ObtenerUser (ObtenerUserRequest) returns (ObtenerUserResponse);
  rpc ObtenerUsers (ObtenerUserRequest) returns (ObtenerUsersResponse);
  rpc EliminarMatricula(EliminarMatriculaRequest) returns (EliminarMatriculaResponse);
}